src/GraphQL/Execution/ArgumentValue.cs

```diff
@@ -0,0 +1,54 @@
+namespace GraphQL.Execution
+{
+    /// <summary>
+    /// Represents the value of an argument.
+    /// </summary>
+    public readonly struct ArgumentValue
+    {
+        /// <summary>
+        /// Initializes a new instance with the specified values.
+        /// </summary>
+        public ArgumentValue(object value, ArgumentSource source)
+        {
+            Value = value;
+            Source = source;
+        }
+
+        /// <summary>
+        /// Returns the value of the argument.
+        /// </summary>
+        public object Value { get; }
```

Comment:
Do I understand correctly that this property cannot contain `null`?

Reply:
No. If a null literal is set for a field argument then it will be null.

Reply:
Also if there is no default for a nullable field and the value is not specified then it will be stored in the dictionary as a null value.

Reply:
Ditto for variable `{"a": null}` ?

Reply:
> Also if there is no default for a nullable field and the value is not specified then it will be stored in the dictionary as a null value.

What `Source` it have in this case?

Reply:
Yes.  Every argument defined on a field is listed in the `Arguments` dictionary, regardless of whether the value was specified, default, etc.

Reply:
Then I only had a question with `Source` property on args with unspecified values.